### ISSUE TITLE
[a11y]: ai-chat should restrict focus to form feedback window and be dismissed with esc key

### ISSUE THREAD

**Issue by Daniel-Schulz-Private** [carbon-design-system/carbon-labs#865]
### Package version

0.18.0

### React version

18.3.1

### Automated testing tool and ruleset

n/a

### Assistive technology

n/a

### Description

Issue description: Unable to work with Modal with standard Keyboard commands :

1. Focus not restricted within form feedback window - Focus coming out and reading from Parent page ( used Tab /Arrow ) 
2. Escape key is not dismissing the form feedback window : Standard keyboard behaviour is missing .

Recommended fix: Make the form feedback window accessible with standard Keyboard /Screen reader commands:
1. Focus is expected to be set to the first Interactive Element with in the form feedback window
2. Focus (both Tab /Arrow ) must be restricted with in the form feedback window.

3. Modal operable with Standard Keyboard commands such as Escape key modal should get Close .
4. After closing the form feedback window - focus should set back to the Triggering control of the Modal in the Parent page.
Note : Detailed Screen reader Testing only possible after making the content identifiable by Screen reader Technologies

User impact: Impact Keyboard /screen reader users :Without proper focus management, users may not realize a modal has opened, leading to confusion, navigation barriers, missed critical information, and a disrupted user experience.

### WCAG 2.1 Violation

WCAG Mapping: 2.4.3 Focus Order

### Reproduction/example

n/a

### Steps to reproduce

1. open form feedback window of ai-chat
2. use focus to navigate the form feedback window

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-labs/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon-labs/issues) for duplicate problems
